common/src/main/java/io/pravega/common/util/CollectionHelpers.java

```diff
@@ -61,6 +62,55 @@
         return -1;
     }
 
+    /**
+     * Performs a binary search on the given IndexedMap.
```

Comment:
That's a good idea, but I have no use for it now. I prefer not to add methods that are not used, until they are needed. 

This can also be done now by using `Collections.singleton(key)`.